plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -555,6 +555,11 @@ private BrokerResponseNative submitBrokerQueryJson(ConnectorSession session, Pin
         });
     }
 
+    private String getScheme()
+    {
+        return getControllerUrl().getScheme();
```

Comment:
Does Pinot allow mixing https & http among controllers? Also, it doesn't allow using different scheme between controllers and brokers?

Reply:
@xiangfu0 if you can confirm: from https://docs.pinot.apache.org/operators/tutorials/configuring-tls-ssl it looks like you can have different schemes for the controllers and brokers as well as multiple schemes for the controllers (while migrating).

Reply:
Technically it can, but I doubt users would mix this, as it breaks the purpose of adding security.
I would just use one switch to force users to do the right thing.
During migration, I think there is anyway a downtime for Trino to switch the configs. So I wouldn't worry too much about it.